### ISSUE TITLE
CabochaOutputError when running align_example.py

### ISSUE THREAD

**Issue by bml1g12** [timmahrt/pyJuliusAlign#3]
In order to run the example, I needed to do something that was suggested in another Issue "`In alignFromTextgrid.py, line 30 should be changed into tg = tgio.openTextgrid(join(inputPath, fn)) instead of capitalizing "TextGrid"`" however even after doing this I obtained the following error:

```
(py27) ben@ben-GL502VSK:~/software/pyJulius/test$ python align_example.py 
introduction.csv
こんにちは
Traceback (most recent call last):
  File "align_example.py", line 35, in <module>
    encoding="utf-8")
  File "/home/ben/anaconda3/envs/py27/lib/python2.7/site-packages/pyjulius/alignFromTextgrid.py", line 77, in convertCorpusToKanaAndRomaji
    cabochaPath)
  File "/home/ben/anaconda3/envs/py27/lib/python2.7/site-packages/pyjulius/juliusAlignment.py", line 237, in prepData
    cabochaPath)
  File "/home/ben/anaconda3/envs/py27/lib/python2.7/site-packages/pyjulius/jProcessingSnippet.py", line 274, in getChunkedKana
    kanaList, wordList = jReads(string, cabochaEncoding, cabochaPath)
  File "/home/ben/anaconda3/envs/py27/lib/python2.7/site-packages/pyjulius/jProcessingSnippet.py", line 219, in jReads
    raise CabochaOutputError(xmlStr)
pyjulius.jProcessingSnippet.CabochaOutputError: Unexpected error in cabocha output (possibly a problem with cabocha).  See error:
morph.cpp(181) [mecab_]
 ```

Do you have any advice how to fix this? I notice that I am using "https://github.com/julius-speech/segmentation-kit" which only has a file called `segment_julius.pl` whereas the other Issue and the documentation refer to a `segment_julius4.pl` file - where can I obtain this? Is it an older version of the segmentation kit?  I don't know if that is related to the issue.

**Comment by bml1g12** [comment]
Ah sorry it was an issue with my Cabocha, which in turn was an issue with Mecab which was an issue with its dictionary encoding and finding relevant dictionary files. Trying to get all these dependencies working is driving me mad, so I might have to give up for now on trying to get this working.  If you know any guide for setting up all these dependencies for non-Japanese operating font encoding (i.e. unicode I guess) I would be very grateful.

**Comment by timmahrt** [comment]
So I'm working on updating this library.  I haven't used it for a long time to be honest.  

What problem are you having right now?  I do recall it being a bit painful to setup.  If you're getting a particular stack trace or error message, that could be helpful.

I just ran it again on my machine (not a fresh install) and had an issue with the cabocha encoding.  I thought I installed the utf-8 version but it was actually euc-jp.  Once I changed the encoding string to that in alignFromTextgrid.convertCorpusToKanaAndRomaji, it worked fine.

Also--and this is a problem I've had elsewhere--I had to supply the full path of the sox and cabocha libraries.  That was contrary to the documentation I had written in the readme file.

**Comment by timmahrt** [comment]
Also, I am using segment_julius4.pl.  I'm not sure where I got it--probably outdated by now.  I will try to install the latest version and make sure it works.

**Comment by bml1g12** [comment]
One error I was getting was when I use mecab it simply printed "morph.cpp(181) [mecab_]" with no stack trace.  

I think I need to uninstall all the relevant ubuntu packages and start compiling Mecab, Cabocha etc. again from scratch being very careful to ensure everything is utf-8 and that the correct utf-8 dictionaries are installed; I'm not sure I'll have time to do that soon though.

**Comment by timmahrt** [comment]
Alright, I've refactored the code and made a new release.

There is an existing project called pyjulius which is older than mine, so I renamed my library to pyJuliusAlign.  The new name better reflects its functionality. Sorry for any inconvenience.

The TextGrid -> Textgrid capitalization issue is resolved, python 3 is now supported, and the code is now available on pypi so it can now be installed via `pip install pyjuliusalign`

In 2015 or so, Julius shifted their codebase to github.  My library was tested against the version that is still available at their old website 'http://julius.osdn.jp/en_index.php?q=index-en.html'

If I can find more time to work on this project, I will try to install the more recent version.  At the moment, I cannot guarentee that my python library will work with the current Julius version on github.

'segment_julius4.pl' is also available on the old website.  The new file 'segment_julius.pl' is actually a completely different file that has different inputs and outputs.  If I can get the new codebase working with my system, I will supply an appropriate 'segment_julius.pl' file.
